### ISSUE TITLE
Breaking changes to Extension not noted in 0.20 changelog

### ISSUE THREAD

**Issue by markedwards** [mirumee/ariadne#1107]
Our implementation broke with 0.20 due to custom `Extension` subclasses we implement.  We run WSGI in Django, with an async view.  We (perhaps incorrectly?) implemented the extensions using `Extension`, not `ExtensionSync`, and in the `async def resolve()` override we were calling `await super().resolve(...)`.

This breaks in 0.20, although the exact logic behind the breakage is hard to follow.  But in any case, it breaks because we end up awaiting an object that is not awaitable.

Shouldn't the change from `Extension.resolve()` being async to it being sync be documented as breaking?

**Comment by rafalp** [comment]
> being async to it being sync be documented as breaking?

We don't note "this is breaking change" in our changelog because we assume people will read the release announcements and get the idea "ok, this will break my usecase".

**Comment by markedwards** [comment]
Fair enough, but the changelog says "Removed ExtensionSync. Extension now supports both async and sync contexts.", and I'm not sure how I could deduce from this that the `resolve` method is changing from async to sync.  I had to go though a fairly convoluted debugging process to figure out what changed.

A note saying "Extension.resolve() is now a sync function, please update any subclasses that override it" would have been helpful.

**Comment by rafalp** [comment]
I understand where you are coming from, but situations like this happened once in a while to me no matter what steps I've took to avoid them and there was always somebody coming to me with with recommendations what to do to avoid the situation repeating (but it still repeated), conditioning me to don't optimize for them in projects I am maintaining. 😅

But I'll take a note and will try to be more mindful in future with those. 👍

**Comment by markedwards** [comment]
No problem, but at least there's this thread in case someone else hits this confusion.  Maybe it will help them.

**Comment by rafalp** [comment]
I'll add the note to release announcement that new extensions implementation is a breaking change. Already did in summary in twitter too.

**Comment by rafalp** [comment]
Updated release announcement:

<img width="804" alt="Zrzut ekranu 2023-06-21 o 14 51 39" src="https://github.com/mirumee/ariadne/assets/750553/8c4c54d1-bdb2-4d53-9fcd-2451b21dd1e8">

Thanks!
